server/session.go

```diff
@@ -45,16 +53,27 @@ func closeSession(w http.ResponseWriter, r *http.Request) error {
 	return session.Save(r, w)
 }
 
-func checkOutUser(w http.ResponseWriter, r *http.Request) (bool, error) {
+func isNewSession(w http.ResponseWriter, r *http.Request) (bool, error) {
 	session, err := store.Get(r, sessionName)
 	if err != nil {
 		return false, err
 	}
-	if session.IsNew == true {
+
+	if session.IsNew {
 		session.Options = &sessions.Options{
 			MaxAge: -1,
 		}
-		return false, session.Save(r, w)
+		return true, session.Save(r, w)
+	}
+
+	return false, nil
+}
+
+func getUserID(r *http.Request) (string, error) {
+	session, err := store.Get(r, sessionName)
+	if err != nil {
+		return "", err
 	}
-	return true, nil
+
+	return fmt.Sprint(session.Values[ID]), err
```

Comment:
Are you using Sprint to convert ID to string? I suppose that there should be other way